src/moments.jl

```diff
@@ -161,87 +161,57 @@ end
 
 ##### General central moment
 function _moment2(v::AbstractArray{<:Real}, m::Real; corrected=false)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        s += z * z
-    end
-    varcorrection(n, corrected) * s
+    s = sum(x->abs2(x-m), v, init=zero(m))
```

Comment:
I think we should use an improved `init` value if possible - in many cases the result of `sum(...)` won't be of the same type as `zero(m)`.

Reply:
For instance, we could use
```suggestion
    init = (zero(eltype(v)) - zero(m))^2
    s = sum(x->(x-m)^2, v; init=init)
```